### ISSUE TITLE
1.6 to 1.7 hpa-upgrade failure: timed out waiting for "res-cons-upgrade" to be synced

### ISSUE THREAD

**Issue by monopole** [kubernetes/kubernetes#47758]
[jenkins](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-1-6-1-7-upgrade-master/61#hpa-upgrade)

[history](https://k8s-testgrid.appspot.com/1.6-1.7-upgrade#gce-upgrade-master&include-filter-by-regex=%5EOverall%24%7Chpa%5C-upgrade)

```
Expected error:
    <*errors.errorString | 0xc420b4c030>: {
        s: "error while stopping { ReplicationController}: res-cons-upgrade: timed out waiting for \"res-cons-upgrade\" to be synced",
    }
    error while stopping { ReplicationController}: res-cons-upgrade: timed out waiting for "res-cons-upgrade" to be synced
not to have occurred

k8s.io/kubernetes/test/e2e/common.(*ResourceConsumer).CleanUp(0xc420a2e0a0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/common/autoscaling_utils.go:402 +0x293
k8s.io/kubernetes/test/e2e/upgrades.(*HPAUpgradeTest).Teardown(0x5688130, 0xc4207008c0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/upgrades/horizontal_pod_autoscalers.go:71 +0x55
panic(0x3517000, 0xc421422b40)
	/usr/local/go/src/runtime/panic.go:489 +0x2cf
k8s.io/kubernetes/test/e2e/framework/ginkgowrapper.Fail.func1(0xc420e14b60, 0xd5, 0x51af443, 0x75, 0x50, 0xc420ab8a00, 0x4a9)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/ginkgowrapper/wrapper.go:63 +0x9c
panic(0x302f020, 0xc420b4df40)
	/usr/local/go/src/runtime/panic.go:489 +0x2cf
k8s.io/kubernetes/test/e2e/framework/ginkgowrapper.Fail(0xc420e14b60, 0xd5, 0xc420b4de08, 0x1, 0x1)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/ginkgowrapper/wrapper.go:67 +0x19b
k8s.io/kubernetes/vendor/github.com/onsi/gomega/internal/assertion.(*Assertion).match(0xc421422900, 0x5443e20, 0x56b01f0, 0x0, 0xc421124840, 0x3, 0x3, 0x56b01f0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/onsi/gomega/internal/assertion/assertion.go:69 +0x1ef
k8s.io/kubernetes/vendor/github.com/onsi/gomega/internal/assertion.(*Assertion).NotTo(0xc421422900, 0x5443e20, 0x56b01f0, 0xc421124840, 0x3, 0x3, 0x544fda0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/onsi/gomega/internal/assertion/assertion.go:43 +0xae
k8s.io/kubernetes/test/e2e/framework.ExpectNoErrorWithOffset(0x1, 0x541d620, 0xc4202295a0, 0xc421124840, 0x3, 0x3)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:1935 +0x14f
k8s.io/kubernetes/test/e2e/common.(*ResourceConsumer).WaitForReplicas(0xc420a2e0a0, 0x1)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/common/autoscaling_utils.go:353 +0x1f3
k8s.io/kubernetes/test/e2e/upgrades.(*HPAUpgradeTest).test(0x5688130)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/upgrades/horizontal_pod_autoscalers.go:80 +0x134
k8s.io/kubernetes/test/e2e/upgrades.(*HPAUpgradeTest).Test(0x5688130, 0xc4207008c0, 0xc4214f0000, 0x0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/upgrades/horizontal_pod_autoscalers.go:64 +0xb9
k8s.io/kubernetes/test/e2e.(*chaosMonkeyAdapter).Test(0xc4215f8300, 0xc4212e6380)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/cluster_upgrade.go:205 +0x1ed
k8s.io/kubernetes/test/e2e.(*chaosMonkeyAdapter).Test-fm(0xc4212e6380)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/cluster_upgrade.go:270 +0x34
k8s.io/kubernetes/test/e2e/chaosmonkey.(*chaosmonkey).Do.func1(0xc4212e6380, 0xc4212de100)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/chaosmonkey/chaosmonkey.go:89 +0x76
created by k8s.io/kubernetes/test/e2e/chaosmonkey.(*chaosmonkey).Do
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/chaosmonkey/chaosmonkey.go:90 +0x1c7
```

**Comment by monopole** [comment]
/sig apps
/assign @enisoc 

Possibly related to #47379

**Comment by dchen1107** [comment]
@jszczepkowski I am assigning this to you since it is related to hpa-upgrade test. Please delegate. Thanks!

**Comment by jszczepkowski** [comment]
@DirectXMan12
Solly, can you take a look?

**Comment by enisoc** [comment]
This looks related to the CIDR issue as @monopole suggested. The test failure is because Pods are not getting cleaned up when a RC is scaled to 0. The RC controller appears to take no action, which is likely because of this error logged in `kube-controller-manager.log`:

```
controllermanager.go:181] error starting controllers: failed to mark cidr as occupied: cidr 10.180.3.0/24 is out the range of cluster cidr 10.100.0.0/14
```

**Comment by dchen1107** [comment]
I quickly looked at the issue, it is a dup of #47379. And after we fixed #47379 (see detail at https://github.com/kubernetes/kubernetes/issues/47379#issuecomment-309912047), the test is not failed.
